netlify/netlify-plugin-nextjs
# fix: move locale detection to netlify redirects

### Summary
Currently, locale detection requires the request to reach the handler function, which means it breaks for static or ODB routes. This PR moves the detection into Netlify redirects, so it happens before other redirects are processed.

This PR also refactors the redirects into a different file, because it was getting unwieldy keeping it inside the config helper

### Test plan

1. TODO: create a test site with a static or fallback page at the root
2. Visit the page with a browser locale of "es"
3. Observe a redirect to `/es/`

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

![image](https://user-images.githubusercontent.com/213306/144217702-069a5f99-5f7e-49c4-8087-4785cb8e3624.png)

Fixes #859

### Standard checks:




---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.
